docker-app/qfieldcloud/core/views/projects_views.py

```diff
@@ -94,6 +94,21 @@ def get_queryset(self):
             if include_public_param and include_public_param.lower() == "1":
                 include_public = True
 
+            owner = self.request.query_params.get("owner")
+            if owner:
+                owner_account = User.objects.get(username=owner)
+                projects = projects.filter(owner=owner_account.id)
+                # When filtering projects by owner, include public projects by default
+                if not include_public_param:
+                    include_public = True
+
+            project = self.request.query_params.get("project")
+            if project:
+                projects = projects.filter(name=project)
+                # When filtering projects by project name, include public projects by default
+                if not include_public_param:
+                    include_public = True
```

Comment:
I would not put this logic neither here, not in the previous occurance. It is hard to document and understand. The client should know whether or not they want to search the public projects too.